src/distributed-transaction/timestamp-oracle.md

```diff
@@ -0,0 +1,15 @@
+# Timestamp Oracle
+
+The timestamp oracle plays a significant role in Percolator Transaction model, it is a server that hands out timestamps in strictly increasing order: a property required for correct operation of the snapshot isolation protocol. 
+
+Since every transaction requires contacting the timestamp oracle twice, this service must scale well. The oracle periodically allocates a range of timestamps by writing the highest allocated timestamp to stable storage; given an allocated range of timestamps, the oracle can satisfy future requests strictly from memory. If the oracle restarts, the timestamps will jump forward to the maximum allocated timestamp (but will never go backwards).
```

Comment:
```suggestion
Since every transaction requires contacting the timestamp oracle twice, this service must scale well. The timestamp oracle periodically allocates a range of timestamps by writing the highest allocated timestamp to stable storage; given an allocated range of timestamps, the timestamp oracle can satisfy future requests strictly from memory. If the timestamp oracle restarts, the timestamps will jump forward to the maximum allocated timestamp (never go backwards).
```